src/elements.jl

```diff
@@ -0,0 +1,144 @@
+import PeriodicTable
+
+# Alias to avoid similarity of elements and Element in DFTK module namespace
+periodic_table = PeriodicTable.elements
+
+# Data structure for chemical element and the potential model via which
+# they interact with electrons. A compensating charge background is
+# always assumed.
+
+abstract type AbstractElement end
+
+"""Return the total nuclear charge of an atom type"""
+charge_nuclear(el::AbstractElement) = el.Z
+
+"""Return the total ionic charge of an atom type (nuclear charge - core electrons)"""
+charge_ionic(el::AbstractElement) = error("Implement charge_ionic")
+
+"""Return the number of valence electrons"""
+n_elec_valence(el::AbstractElement) = charge_ionic(el)
+
+"""Return the number of core electrons"""
+n_elec_core(el::AbstractElement) = charge_nuclear(el) - charge_ionic(el)
+
+function local_potential_fourier(el::AbstractElement, q::AbstractVector)
+    local_potential_fourier(el, norm(q))
+end
+
+function local_potential_real(el::AbstractElement, q::AbstractVector)
+    local_potential_real(el, norm(q))
+end
+
+
+"""
+Element interacting with electrons via a bare Coulomb potential
+(for all-electron calculations)
+"""
+struct ElementAllElectron <: AbstractElement
+    Z::Int  # Nuclear charge
+    symbol  # Element symbol
+end
+charge_ionic(el::ElementAllElectron) = el.Z
+
+function ElementAllElectron(key)
```

Comment:
What is key? String, symbol?

Reply:
Actually all of them. I'll add some docs.